pallets/phala/src/stakepool.rs

```diff
@@ -132,26 +143,41 @@ pub mod pallet {
 				.as_secs()
 				.saturated_into::<u64>();
 
-			// TODO:
-			// 1) should we just stop some of workers rather than all of it
-			// 2) just iterate pools that contains waitting withdraw rather than all of it
-			for pid in 0..PoolCount::<T>::get() {
-				let pool_info = Self::ensure_pool(pid).expect("Stake pool doesn't exist; qed.");
-				if !pool_info.withdraw_queue.is_empty() {
-					// the front withdraw always the oldest one
-					if let Some(info) = pool_info.withdraw_queue.front() {
-						if (now - info.start_time)
-							> T::InsurancePeriod::get().saturated_into::<u64>()
-						{
-							// stop all worker in this pool
-							// TODO: only stop running workers?
-							for worker in pool_info.workers {
-								let miner: T::AccountId = pool_sub_account(pid, &worker);
-								let _ = <mining::pallet::Pallet<T>>::stop_mining(miner);
+			let mut t = WithdrawTimestamps::<T>::get();
+			// has waiting withdraw request
+			if !t.is_empty() {
```

Comment:
I suggest to use the following pattern to iterate the queue. It can generally avoid multiple `empyt()`, `front().unwrap()`, and `pop_front()` calls.

```rust
while let Some(start_time) = t.pop_front() {
}
```